### ISSUE TITLE
Advise on hyper param change

### ISSUE THREAD

**Issue by skol101** [yl4579/StarGANv2-VC#36]
Hello,

I've trained for 165 epochs, but eval/cyc stays about the same after step 50. 

Should I train with a new config where adv_cls_epoch and con_reg_epoch starts later? 
![Screenshot from 2022-02-16 15-06-59](https://user-images.githubusercontent.com/53978091/154270854-87929965-bde2-4908-84f5-480bfc19237a.png)

**Comment by skol101** [comment]
For instance, will batch size (5 in provided config vs 32 for rtx 3090) have any impact on convergence?

**Comment by yl4579** [comment]
The hyperparameters barely have any effects in my experience, those were chosen arbitrarily and could possibly be anything reasonable (e.g., as long as you don't start training adversarial classifiers or data augmentation right away). The cycle loss might seem to have converged, but again you should keep training until it starts to increase because in GAN training what really matters is not the loss values but the actual quality of generated images (or speech in our case). Loss values are generally not interpretable in GAN training, you will need to load different checkpoints and assess the performance subjectively to decide which one to use.

As for the batch size, a larger batch size might cause the discriminator to dominate because they see more samples and could result in worse generator performance, that's why in GAN training the batch size is usually set to be small. You can try to change the lambda for the adversarial loss though to make the training balanced.
